components/BannerManager/Liquidation/useLiquidationOptimismSubgraph.ts

```diff
@@ -52,30 +46,41 @@ export async function fetchLiquidationInfo(
 		}),
 	});
 
-	const { data, errors } = await body.json();
+	const {
+		errors,
+		data,
+	}: {
+		errors?: Error[];
+		data: { stakerEntity: StakerEntity };
+	} = await body.json();
```

Comment:
> ok, that's where I was missing another StakerEntity... wow

Yea and making sure this fetch fn have a proper type. By default this will be `any`. The proper way of doing it would be to rely on a runtime validation library like zod. But this works alright if the API is stable

Reply:
Yea. The generic is needed if you want consumers of the hook to pass in options. If you have the " inner callback fetch function" typed and have the options configured directly in your hook you dont need the generic at all.